File: code_blocks/virtual-currency/vc-balance-sdk-android.java
// Fetch virtual currencies
Purchases.sharedInstance.getVirtualCurrencies(
    new GetVirtualCurrenciesCallback() {
        @Override
        public void onReceived(@NonNull VirtualCurrencies virtualCurrencies) {
            // TODO: Handle virtual currencies
        }

        @Override
        public void onError(@NonNull PurchasesError error) {
            // TODO: Handle error
        }
    }
);

// Get the details of a specific virtual currency
VirtualCurrency virtualCurrency = virtualCurrencies.getAll().get(<virtual_currency_code>);
int balance = virtualCurrency.getBalance();
String name = virtualCurrency.getName();
String code = virtualCurrency.getCode();

// Keep in mind that serverDescription may be null if no description was provided
// in the RevenueCat dashboard
String serverDescription = virtualCurrency.getServerDescription();

// Iterate through all virtual currencies
virtualCurrencies.getAll().forEach((key, virtualCurrency) -> {
    System.out.println(virtualCurrency.getCode() + ": " + virtualCurrency.getBalance());
});

File: code_blocks/virtual-currency/vc-balance-sdk-android.kt
// Fetch virtual currencies
// With coroutines
try {
    val getVirtualCurrenciesResult: VirtualCurrencies = Purchases.sharedInstance.awaitGetVirtualCurrencies()
    // TODO: Handle virtual currencies
} catch(error: PurchasesException) {
    // TODO: Handle error
}

// With callbacks
Purchases.sharedInstance.getVirtualCurrenciesWith(
    onError = { error: PurchasesError -> 
        // TODO: Handle error
    },
    onSuccess = { virtualCurrencies: VirtualCurrencies -> 
        // TODO: Handle virtual currencies
    },
)

// Get the details of a specific virtual currency
val virtualCurrency = virtualCurrencies.all[<your_virtual_currency_code>]
val balance = virtualCurrency?.balance
val name = virtualCurrency?.name
val code = virtualCurrency?.code

// Keep in mind that serverDescription may be null if no description was provided
// in the RevenueCat dashboard
val description = virtualCurrency?.serverDescription

// Iterate through all virtual currencies
for ((virtualCurrencyCode, virtualCurrency) in virtualCurrencies.all) {
    println("$virtualCurrencyCode: ${virtualCurrency.balance}")
}